app/src/lib/rules/checker_rules.js

```diff
@@ -404,16 +435,40 @@ export const CHECKER_RULES = builtin_checker_rules.map(({ rule }) => rule).conca
 
 /**
  * 
- * @param {OntologyResult} result 
+ * @param {(result: OntologyResult?) => boolean} level_check 
+ * @returns {TokenFilter}
+ */
+function check_token_level(level_check) {
+	return token => {
+		return token.lookup_results.length > 0
+			&& token.lookup_results.every(result => level_check(result.concept))
+	}
+}
+/**
+ * 
+ * @param {(result: OntologyResult?) => boolean} level_check 
+ * @returns {TokenFilter}
+ */
+function check_ambiguous_level(level_check) {
+	return token => {
+		return token.lookup_results.length > 0
+			&& level_check(token.lookup_results[0].concept)
+			&& token.lookup_results.some(result => !level_check(result.concept))
+	}
+}
+
+/**
+ * 
+ * @param {OntologyResult?} result 
  */
 function is_level_simple(result) {
-	return result.level === 0 || result.level === 1
+	return result !== null && (result.level === 0 || result.level === 1)
 }
 
 /**
  * 
- * @param {OntologyResult} result 
+ * @param {OntologyResult?} result 
  */
 function is_level_complex(result) {
-	return result.level === 2 || result.level === 3
+	return result !== null && (result.level === 2 || result.level === 3)
```

Comment:
```suggestion
	return [2, 3].includes(result?.level)
```

this might also work.

Reply:
`includes()` is type checked, so doesn't like the possible null argument. I'll use the `includes()` but keep the null check

Reply:
oh ok, bummer